gromacs/tools.py

```diff
@@ -94,177 +108,262 @@ def _generate_sphinx_class_string(clsname):
+    return tools
+
+
+def load_extra_tools():
+    """ Load extra tools.
+
+    :return: dict mapping tool names to GromacsCommand classes
+    """
+    names = config.cfg.get("Gromacs", "extra").split()
+    tools = {}
+    for name in names:
+        fancy = make_valid_identifier(name)
+        driver = 'gmx' if config.MAJOR_RELEASE == '5' else None
+        tools[name] = tool_factory(fancy, name, driver)
+    return tools
+
+
+def merge_ndx(*args):
+    """ Takes one or more index files and optionally one structure file and
+    returns a path for a new merged index file.
+
+    :param args: index files and zero or one structure file
+    :return: path for the new merged index file
+    """
+    ndxs = []
+    struct = None
+    for fname in args:
+        if fname.endswith('.ndx'):
+            ndxs.append(fname)
+        else:
+            assert struct is None, "only one structure file supported"
+            struct = fname
+
+    fd, multi_ndx = tempfile.mkstemp(suffix='.ndx', prefix='multi_')
+    atexit.register(os.unlink, multi_ndx)
+
+    if struct:
+        make_ndx = registry['Make_ndx'](f=struct, n=ndxs, o=multi_ndx)
+    else:
+        make_ndx = registry['Make_ndx'](n=ndxs, o=multi_ndx)
+
+    _, _, _ = make_ndx(input=['q'], stdout=False, stderr=False)
+    return multi_ndx
+
+
+if config.MAJOR_RELEASE == '5':
+    registry = load_v5_tools()
+elif config.MAJOR_RELEASE == '4':
+    registry = load_v4_tools()
+else:
+    try:
+        registry = load_v5_tools()
+    except GromacsToolLoadingError:
+        try:
+            registry = load_v4_tools()
+        except GromacsToolLoadingError:
+            raise GromacsToolLoadingError("Unable to load any tool")
+
+registry.update(load_extra_tools())
+
+
+# Aliases command names to run unmodified GromacsWrapper scripts on a machine
+# with only 5.x
+for fancy, cmd in registry.items():
+    for c5, c4 in NAMES5TO4.iteritems():
+        # have to check each one, since it's possible there are suffixes
+        # like for double precision
+        name = cmd.command_name
+        if name.startswith(c5):
+            if c4 == c5:
+                break
+            else:
+                # mantain suffix
+                name = c4 + fancy.lower().split(c5)[1]
+                registry[make_valid_identifier(name)] = registry[fancy]
+                break
+    else:
+        # the common case of just adding the 'g_'
+        registry['G_%s' % fancy.lower()] = registry[fancy]
+
+for name4, name5 in [('G_mindist', 'Mindist'), ('G_dist', 'Distance')]:
+    if name4 in registry:
+        klass = type(name4, (GromacsCommandMultiIndex,),  {
+            'command_name': registry[name5].command_name,
+            'driver': registry[name5].driver,
+            '__doc__': registry[name5].__doc__
+        })
+        registry[name4] = klass
+        if name5 in registry == '5':
+            registry[name5] = klass
 
 
 # 5.0.5 compatibility hack
 if 'Convert_tpr' in registry:
```

Comment:
Stupid question... I can't remember why we are doing this here. Can't we just include that in the aliases?